File: SDKs/amazonaws.com/appflow/2020-08-23/java/lib/src/main/java/openapisdk/models/shared/ZendeskConnectorProfileCredentials.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ZendeskConnectorProfileCredentials
 *  The connector-specific profile credentials required when using Zendesk. 
**/
public class ZendeskConnectorProfileCredentials {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("accessToken")
    public String accessToken;
    public ZendeskConnectorProfileCredentials withAccessToken(String accessToken) {
        this.accessToken = accessToken;
        return this;
    }
    @JsonProperty("clientId")
    public String clientId;
    public ZendeskConnectorProfileCredentials withClientId(String clientId) {
        this.clientId = clientId;
        return this;
    }
    @JsonProperty("clientSecret")
    public String clientSecret;
    public ZendeskConnectorProfileCredentials withClientSecret(String clientSecret) {
        this.clientSecret = clientSecret;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("oAuthRequest")
    public ConnectorOAuthRequest oAuthRequest;
    public ZendeskConnectorProfileCredentials withOAuthRequest(ConnectorOAuthRequest oAuthRequest) {
        this.oAuthRequest = oAuthRequest;
        return this;
    }
}

File: SDKs/amazonaws.com/appflow/2020-08-23/python/src/sdk/models/shared/zendeskconnectorprofilecredentials.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import connectoroauthrequest as shared_connectoroauthrequest


@dataclass_json
@dataclasses.dataclass
class ZendeskConnectorProfileCredentials:
    r"""ZendeskConnectorProfileCredentials
     The connector-specific profile credentials required when using Zendesk. 
    """
    
    client_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('clientId') }})
    client_secret: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('clientSecret') }})
    access_token: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accessToken') }})
    o_auth_request: Optional[shared_connectoroauthrequest.ConnectorOAuthRequest] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('oAuthRequest') }})
    
